core/arch/arm/kernel/kern.ld.S

```diff
@@ -261,13 +283,24 @@ SECTIONS
 		__start_phys_sdp_mem_section = . ;
 		KEEP(*(phys_sdp_mem_section))
 		__end_phys_sdp_mem_section = . ;
-		. = ALIGN(8);
+
+		/*
+		 * Insure init sections ends on mapping page
```

Comment:
Again I understand your concern about physical memory waste in paged configuration. 

I will propose a `CFG_CORE_RODATA_NONEXECUTABLE=y|n`.

One way to overcome this specific alignment could be to swap paged executable text and paged read-only-data. Would you agree with this change ?